### ISSUE TITLE
Legion Prof buggy with spy data

### ISSUE THREAD

**Issue by nsccap** [StanfordLegion/legion#753]
Summary: changes in legion_spy.py broke legion_proj for cases with spy data

What version: latest stable legion-19.12.0

What happens: "legion_prof.py ./prof_0.gz ./spy_0.log" crashes in multiple places. This seems
to be caused by the following commit that changed a few critical things that prof relied on:
```
commit 91b65bd67c74a0c8bfa034d3da62bafdf774e0c2
Author: Mike <mebauer@cs.stanford.edu>
Date:   Thu Sep 26 16:26:12 2019 -0700

    tools: updates to legion spy to dump bad equivalence set graphs
```

I fixed (cludged) it up enough to run locally but I'm pretty sure it's to ugly for a PR. In short API
change in two places (diff output trimmed for brevity):
```
--- a/tools/legion_prof.py
+++ b/tools/legion_prof.py
@@ -3283,7 +3283,7 @@ class State(object):
-        self.spy_state = legion_spy.State(False, True, True, True)
+        self.spy_state = legion_spy.State("/tmp", False, True, True, True, False)
-        self.spy_state.perform_physical_analysis(False, False, False)
+        self.spy_state.perform_physical_analysis(False, False)
```

And a bit later it failed to notice the root context so I added explicit check:

```
@@ -3398,6 +3398,9 @@ class State(object):
         for op in itervalues(self.spy_state.ops):
             if op.context is not None:
+                if str(op.context) == "Root context":
+                    continue
```

After that the example I ran produced a seemingly ok output directory that rendered in the browser.

**Comment by seemamirch** [comment]
This has been fixed in the master branch and will be merged into stable in the next release

On Feb 24, 2020, at 9:24 AM, Peter Kjellstrom <notifications@github.com<mailto:notifications@github.com>> wrote:


Summary: changes in legion_spy.py broke legion_proj for cases with spy data

What version: latest stable legion-19.12.0

What happens: "legion_prof.py ./prof_0.gz ./spy_0.log" crashes in multiple places. This seems
to be caused by the following commit that changed a few critical things that prof relied on:

commit 91b65bd67c74a0c8bfa034d3da62bafdf774e0c2
Author: Mike <mebauer@cs.stanford.edu<mailto:mebauer@cs.stanford.edu>>
Date:   Thu Sep 26 16:26:12 2019 -0700

    tools: updates to legion spy to dump bad equivalence set graphs


I fixed (cludged) it up enough to run locally but I'm pretty sure it's to ugly for a PR. In short API
change in two places (diff output trimmed for brevity):

--- a/tools/legion_prof.py
+++ b/tools/legion_prof.py
@@ -3283,7 +3283,7 @@ class State(object):
-        self.spy_state = legion_spy.State(False, True, True, True)
+        self.spy_state = legion_spy.State("/tmp", False, True, True, True, False)
-        self.spy_state.perform_physical_analysis(False, False, False)
+        self.spy_state.perform_physical_analysis(False, False)


And a bit later it failed to notice the root context so I added explicit check:

@@ -3398,6 +3398,9 @@ class State(object):
         for op in itervalues(self.spy_state.ops):
             if op.context is not None:
+                if str(op.context) == "Root context":
+                    continue


After that the example I ran produced a seemingly ok output directory that rendered in the browser.
